t/kvs/watch_disconnect.c

```diff
@@ -66,15 +82,31 @@ int main (int argc, char **argv)
         log_err_exit ("flux_open");
     w0 = count_watchers (h);
     send_watch_requests (h, "nonexist");
-    w1 = count_watchers (h) - w0;
+
+    /* must spin / wait for watchers to be registered */
+    for (i = 0; i < MAX_ITERS; i++) {
+        w1 = count_watchers (h) - w0;
+        if (w1 == rankcount)
```

Comment:
could the test use a barrier here instead of looping and sleeping? I guess if the `sleep (1)` is only rarely invoked it isn't too big a deal...

Reply:
the test was passing all the time in the past, so hopefully the sleeps are rare :-)

Not sure a barrier will be doable.  Basically, they are spins to ensure that RPCs have been sent / received / processed b/c in both cases an RPC response isn't coming back.  Perhaps should use ```usleep()``` instead?

Reply:
If the sleep is rare then it isn't too big of a deal.

Reply:
I sleep "0.1" seconds in various kvs tests, so I'll just continue the trend and convert to usleep in 0.1 second chunks.